docs/bos/queryapi/context.md

```diff
@@ -0,0 +1,348 @@
+
+#### Input
+
+```js
+await context.set(key, value)
+```
+
+#### Example
+
+```js
+const h = block.header().height;
+await context.set("height", h);
+```
+
+### Log
+
+Code written by developers is executed on a GCP compute instance. Therefore, things like `console.log` are surfaced to the machine itself. So, to surface `console.log` statements not only to the machine but also back to the user under Indexer Status, we need to write these logs to the logging table, which is separate from the developer's schema. This actually happens in the runner. We map `console.log` statements in the developer's code to call this function instead. So, for the developer, they can simply use `console.log`, and we take care of the rest.
+
+
+### FetchFromSocialApi
+
+Calls to APIs are restricted in QueryAPI because making calls usually require establishing identity. QueryAPI does not support secrets yet (e.g., identity tokens) as those would end up stored on the blockchain, which is not secure.
+However, social API doesn’t require a secret, and has some potential uses. So, QueryAPI supports calls to it through this method. 
+
+#### Input
+
+```js
+await context.fetchFromSocialApi(path, options)
+```
+
+The path is a resource from the social API that the developer is targeting. Options are where the rest of the call is placed (`headers`, `method`, `body`, and so on). 
+
+#### Example
+
+```js
+const response = await context.fetchFromSocialApi("/index", {
+  method: "POST",
+  headers: {
+    ["Content-Type"]: "application/json",
+  },
+  body: JSON.stringify({
+    action: "post",
+    key: "main",
+    options: {
+    limit: 1,
+    order: "desc",
+  },
+  }),
+});
+```
+
+:::tip
+
+This snippet is from the [social_lag indexer](https://near.org/dev-queryapi.dataplatform.near/widget/QueryApi.App?selectedIndexerPath=morgs.near/social_lag&view=editor-window). 
+
+:::
+
+## DB
+
+The DB object is a sub-object under `context`. It is accessed through `context.db`. Previously, the GraphQL method was the only way to interact with the database. However, writing GraphQL queries and mutations is pretty complicated and overkill for simple interactions. So, simpler interactions are instead made available through the db sub-object. This object is built by reading the schema written by the developer, parsing its information, and generating methods for each table. See below for what methods are generated for each table. The format to access the below methods is as follows: `context.db.[tableName].[methodName]`. Concrete examples are also given below. 
+
+:::info Note
+
+One thing to note is that the process where the code is read is not fully featured. If `ALTER TABLE ALTER COLUMN` is used, for example, in the schema, it will fail to parse. Should this failure occur, the context object will still be generated but `db` methods will be unavailable. An error will appear on the developer’s page saying types were not generated. 
```

Comment:
We can rewrite the second line to: 
If an `ALTER TABLE ALTER COLUMN` statement is used in the SQL schema, for example, it will fail to parse.

Also, like mentioned in my previous comment, lines like 
`An error will appear on the developer’s page saying types were not generated`
 should be rewritten to not talk as though the customer is not the one reading it. That line should be something like 
`An error will appear on the page saying types could not be generated. A more detailed error can be viewed in the browser console.`

Reply:
fixed